hack/lib/test/junit.sh

```diff
@@ -186,7 +186,7 @@ function os::test::junit::internal::generate_report() {
     os::util::ensure::built_binary_exists 'junitreport'
 
     local report_file
-    report_file="$( mktemp --tmpdir="${ARTIFACT_DIR}" "${report_type}_report_XXXXX" --suffix ".xml" )"
+    report_file="$( mktemp "${ARTIFACT_DIR}/${report_type}_report_XXXXX" ).xml"
```

Comment:
This is a "your mktemp flags don't work anywhere else".  I don't know that the extra flags are any better than this.